messaging/WFLY-6660_artemis_data_tools.adoc

```diff
@@ -0,0 +1,89 @@
+= Allow to dump Artemis journal to file using WildfFly Management API
+:author:            Emmanuel Hugonnet
+:email:             ehugonne@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:          jms,messaging
+:idseparator:       -
+
+== Overview
+
+For debugging and investigation purposes it would useful to have a WildFly management API operation which would dump content of Artemis journal directory.
+This new operation would print content of journal to a file. It would take advantage of https://activemq.apache.org/components/artemis/documentation/latest/data-tools.html[Artemis data tools print command].
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-6660[WFLY-6660]
+
+=== Related Issues
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+The new operation 'data-print' will added  to the https://wildscribe.github.io/WildFly/21.0/subsystem/messaging-activemq/server/[/subsystem=messaging-activemq/server resource]. It will print a report about journal records and summary of existent records, as well a report on paging. This operation shouldn't be executed against a running broker.
+
+=== Hard Requirements
+
+Provide a new 'data-print' operation that will return a stream with the following parameters:
+ * safe a Boolean parameter with a default value of 'false' to print the data structure without showing the your data.
```

Comment:
I'm not sure what this parameter is meant to do. Could you provide more details. please?